pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -96,6 +99,13 @@ func InitializeSecurityPolicy(service common.Service, vpcService common.VPCServi
 		}),
 		BindingType: model.RuleBindingType(),
 	}}
+
+	securityPolicyService.projectGroupStore = &GroupStore{ResourceStore: common.ResourceStore{
```

Comment:
I introduce one more group store specific for project group is to simplify the logic and save running time.Space consumption should be the same for using one store and two stores. 
If without project store, in the logic blow:
https://github.com/vmware-tanzu/nsx-operator/pull/490/files/a2d5bf2cfe22b4cc87339447bcf8a2eacd1d839a#diff-53bc6d3b74f0d34b4187204c1bcf7ce4afe93c7f73e60b5de43c40dc44b88af0R417

and in the firstly, we need to build new project groups with buildSecurityPolicy
nsxSecurityPolicy, nsxGroups, nsxProjectShares, err := service.buildSecurityPolicy(obj, createdFor).

and then,  we have to find all existing project groups from group store using groupShared tag, like below loop:

for i := len(groups) - 1; i >= 0; i-- {
	for j := len(groups[i].Tags) - 1; j >= 0; j-- {
			if *(groups[i].Tags[j].Scope) == common.TagScopeProjectGroupShared {
				if *(groups[i].Tags[j].Tag) == "true" {
					nsxProjectGroups = append(nsxProjectGroups, *groups[i])
                               }
                      }
           }
}

finally, we can compare these project groups if there is any change:
changed, stale := common.CompareResources(GroupsPtrToComparable(existingProjectGroups), GroupsToComparable(projectGroups))

This project is taking more time and more complicated, that is why I add one more project group store. Of course, bringing one more store also brings more store management code, but, it helps to simply the project group update logic because we need to handle project group in the  API other than the groups saved in VPC level.